flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/BufferResponseAndNoDataBufferMessageParser.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.netty;
+
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBuf;
+
+/**
+ * The parser for both {@link NettyMessage.BufferResponse} and messages without data buffer.
+ */
+public class BufferResponseAndNoDataBufferMessageParser extends NoDataBufferMessageParser {
```

Comment:
The definition/naming of  `BufferResponseAndNoDataBufferMessageParser` and `NoDataBufferMessageParser` seems ambiguous. One is used for server/sender side and another is used client/receiver side by design, but it is also divided by whether it has `BufferResponse` in practice. So it is not very clear to know the criteria to distinguish.

As we confirmed before, we can focus on the received side for this PR and remove the `NoDataBufferMessageParser`. I guess we might even remove this parser component completely because the `getMessageHeaderLength` method is also not invalid as I comment below.

Reply:
I have removed the different message parsers according to the previous discussion. Now `NettyMessageParser` becomes a concrete tool class with package visibility. I think we could future discuss if it is better to merge this class directly to `ZeroMessageNettyMessageDecoder`.